pkg/controller/smart-clone-controller.go

```diff
@@ -335,16 +343,30 @@ func (c *SmartCloneController) updateSmartCloneStatusPhase(phase cdiv1.DataVolum
 		event.message = fmt.Sprintf(MessageCloneSucceeded, dataVolumeCopy.Spec.Source.PVC.Namespace, dataVolumeCopy.Spec.Source.PVC.Name, newPVC.Namespace, newPVC.Name)
 	}
 
-	return c.emitEvent(dataVolume, dataVolumeCopy, &event)
+	return c.emitEvent(dataVolume, dataVolumeCopy, &event, newPVC)
 }
 
-func (c *SmartCloneController) emitEvent(dataVolume *cdiv1.DataVolume, dataVolumeCopy *cdiv1.DataVolume, event *DataVolumeEvent) error {
+func (c *SmartCloneController) emitEvent(dataVolume *cdiv1.DataVolume, dataVolumeCopy *cdiv1.DataVolume, event *DataVolumeEvent, newPVC *corev1.PersistentVolumeClaim) error {
 	// Only update the object if something actually changed in the status.
 	if !reflect.DeepEqual(dataVolume.Status, dataVolumeCopy.Status) {
 		_, err := c.cdiClientSet.CdiV1alpha1().DataVolumes(dataVolume.Namespace).Update(dataVolumeCopy)
-		// Emit the event only when the status change happens, not every time
-		if event.eventType != "" {
-			c.recorder.Event(dataVolume, event.eventType, event.reason, event.message)
+		if err != nil {
+			if k8serrors.IsConflict(err) {
+				// Something else updated before we could. Refresh data volume and try again
+				klog.V(3).Info("Something updated the data volume, fetching fresh one and trying again.")
+				refreshDataVolume, err := c.dataVolumeLister.DataVolumes(dataVolume.Namespace).Get(dataVolume.Name)
+				if err != nil {
+					klog.Errorf("error getting data volume: %v", err)
+					return err
+				}
+				return c.updateSmartCloneStatusPhase(dataVolumeCopy.Status.Phase, refreshDataVolume, newPVC)
```

Comment:
emitEvent is called by `updateSmartCloneStatusPhase` so recursing here.  Forever it keeps on failing.

Reply:
Yes I debated putting a limit on the recursion here. Btw it only recurses IF the error returned is a conflict error. So it would only recurse forever if each time in between us refreshing the object and attempting to write the update, something else has updated the object.

Reply:
Removed this, and added a c.queue.Add instead, so no recursion.